src/Interpreters/Session.cpp

```diff
@@ -182,7 +182,7 @@ class NamedSessionsStorage
     std::chrono::steady_clock::time_point close_cycle_time = std::chrono::steady_clock::now();
     UInt64 close_cycle = 0;
 
-    void scheduleCloseSession(NamedSessionData & session, std::unique_lock<std::mutex> &)
```

Comment:
Consider using [TSA](https://github.com/ClickHouse/ClickHouse/blob/20cd799e518c25e41de9cf6ce1bec1429fda2749/base/base/defines.h#L130-L152)

Reply:
Thanks for advicing to use TSA, 

It helped to find a race condition on named sessions,
[std::unique_lock](https://en.cppreference.com/w/cpp/thread/unique_lock) was used, but never locked. It seems we need to use lock_guard

Reply:
Unfortunally, It may be some kind of Sanitizer error, at least i don't get why the error happens in this context:

![unique_lock](https://github.com/ClickHouse/ClickHouse/assets/16054347/bd70639d-cfc5-4575-a0e3-0f1b9464bec4)

Error text:
```
/home/admin/ClickHouse5/ClickHouse/src/Interpreters/Session.cpp:212:29: error: calling function 'closeSessions' requires holding mutex 'mutex' exclusively [-Werror,-Wthread-safety-analysis]
            auto interval = closeSessions();
                            ^
/home/admin/ClickHouse5/ClickHouse/src/Interpreters/Session.cpp:207:5: note: thread warning in function 'cleanThread'
    {
    ^
1 error generated.
```

Reply:
I just reverted my changes, and keep unused `unique_lock` in parameters.
But later I found this:
https://clang.llvm.org/docs/ThreadSafetyAnalysis.html

To use unique_lock with TSA, we need to use `PT_GUARDED_BY ` macro in Clang